rio_tiler/io/stac.py

```diff
@@ -224,4 +229,11 @@ def _get_asset_url(self, asset: str) -> str:
         if asset not in self.assets:
             raise InvalidAssetName(f"{asset} is not valid")
 
-        return self.item.assets[asset].get_absolute_href()
+        asset_info = self.item.assets[asset]
+        info = AssetInfo(url=asset_info.get_absolute_href())
+
+        if "file:header_size" in asset_info.extra_fields:
+            h = asset_info.extra_fields["file:header_size"]
+            info["env"] = {"GDAL_INGESTED_BYTES_AT_OPEN": 16384 * math.ceil(h / 16384)}
```

Comment:
✨

Reply:
Isn't this the same as `{"GDAL_INGESTED_BYTES_AT_OPEN": h}`

Reply:
I guess it should